compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -545,20 +545,26 @@ class ClassfileParser(
       case CLASS_TAG =>
         if (skip) None else Some(lit(Constant(pool.getType(index))))
       case ENUM_TAG =>
-        val enumClassTp = pool.getType(index)
+        val sig = pool.getExternalName(index).value
         val enumCaseName = pool.getName(in.nextChar)
         if (skip)
           None
         else {
-          val enumModuleClass = enumClassTp.classSymbol.companionModule
-          Some(Select(ref(enumModuleClass), enumCaseName.name))
+          val res = (ctx: Context) => {
+            implicit val ctx2 = ctx
+            val enumClassTp = sigToType(sig)
```

Comment:
Could we store just `sig` instead (so this method would return `Option[untpd.Tree | String]`) ?

Reply:
Yes, that's a good idea, I'll try that.